### PR TITLE
Worker should contribute when app/data_downloa_failed & refactor replicate update

### PR DESCRIPTION
**Pull request by zied-guesmi**

### CODE REVIEW

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java [hunk 1]
```diff
@@ -216,153 +216,187 @@ public void updateReplicateStatus(String chainTaskId,
...
     }
 
-    private String getReplicateStatusSuccessLog(String chainTaskId, String walletAddress, ReplicateStatus currentStatus, ReplicateStatus newStatus, ReplicateStatusModifier modifier) {
-        return getUpdateReplicateStatusLogMessage("UpdateReplicateStatus succeeded", "", chainTaskId, walletAddress, currentStatus, newStatus, modifier);
+    private boolean verifyStatus(String chainTaskId, String walletAddress, ReplicateStatus status) {
+        switch (status) {
+            case CONTRIBUTED:
+                return iexecHubService.isStatusTrueOnChain(chainTaskId, walletAddress,
+                        ChainContributionStatus.CONTRIBUTED);
+            case CONTRIBUTE_FAILED:
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/replicate/ReplicatesService.java]: I think we don't want to update to CONTRIBUTE_FAILED & REVEAL_FAILED anymore

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java [hunk 2]
```diff
@@ -216,153 +216,187 @@ public void updateReplicateStatus(String chainTaskId,
...
+            log.error("Cannot update replicate, bad wokfow transition {}",
+                    getLogDetails(chainTaskId, replicate, statusUpdate));
             return Optional.empty();
         }
 
-        replicate = updateReplicateFields(chainTaskId, replicate, statusUpdate);
+        boolean canUpdate = true;
+
+        switch (newStatus) {
+            case CONTRIBUTE_FAILED:
```
Comment (jeremyjams) [inline src/main/java/com/iexec/core/replicate/ReplicatesService.java]: We don't want this status anymore

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java (hunk 1 above)
Comment (zied-guesmi) [reply src/main/java/com/iexec/core/replicate/ReplicatesService.java]: Why not? the worker may not succeed to contribute or reveal (tx issues or not correct hash)!

Comment (jeremyjams) [reply src/main/java/com/iexec/core/replicate/ReplicatesService.java]: As a worker, are you able to provide me a proof of a failed contribution?
If yes, so we can keep this status

Comment (zied-guesmi) [reply src/main/java/com/iexec/core/replicate/ReplicatesService.java]: Well it's the same also for the result upload!

File: src/main/java/com/iexec/core/replicate/ReplicatesService.java (hunk 2 above)
Comment (Ugo) [reply src/main/java/com/iexec/core/replicate/ReplicatesService.java]: As discussed, we can include CONTRIBUTE_FAILED and REVEAL_FAILED but there should be no update performed on the replicate.

**Pull request merged**